Extractor/RebrowseTriggerManager.cs

```diff
@@ -0,0 +1,151 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+using Cognite.OpcUa.History;
+using Microsoft.Extensions.Logging;
+
+using Opc.Ua;
+using Opc.Ua.Client;
+
+namespace Cognite.OpcUa
+{
+    public class RebrowseTriggerManager
+    {
+        private readonly ILogger<RebrowseTriggerManager> logger;
+        private readonly UAClient _uaClient;
+        private readonly RebrowseTriggersConfig _config;
+        private readonly UAExtractor _extractor;
+
+        public readonly static  string SubscriptionName = "TriggerRebrowse";
+
+        public RebrowseTriggerManager(
+            ILogger<RebrowseTriggerManager> logger,
+            UAClient uaClient,
+            RebrowseTriggersConfig config,
+            UAExtractor extractor
+        )
+        {
+            this.logger = logger;
+            _uaClient = uaClient;
+            _config = config;
+            _extractor = extractor;
+        }
+
+        public async Task EnableCustomServerSubscriptions(CancellationToken token)
+        {
+            var targetNodes = _config.Targets.GetValues;
+            List<NodeId> nodeIds = new List<NodeId>();
+            var filteredNamespaces = _config.Namespaces;
+            var filteredNamespacesCount = filteredNamespaces?.Count();
+            var shouldFilterNamespaces = filteredNamespacesCount > 0;
+
+            var serverNamespaces = ObjectIds.Server_Namespaces;
+
+            var grouping = new Dictionary<NodeId, List<ReferenceDescription>>();
+            // displayName: nodeId
+            var namespaceNameToId = new Dictionary<string, string>();
+
+            await _uaClient.Browser.BrowseDirectory(
+                new[] { serverNamespaces },
+                (refDef, parent) =>
+                {
+                    var nodeId = (NodeId)refDef.NodeId;
+
+                    if (parent == serverNamespaces && !grouping.ContainsKey(nodeId))
+                    {
+                        grouping.Add(nodeId, new List<ReferenceDescription>());
+                        namespaceNameToId.Add(refDef.DisplayName.ToString(), nodeId.ToString());
```

Comment:
I'd rather we don't use NodeId.ToString(): Is there are reason why you need it here? NodeId implements equality properly, so you can just use the `NodeId` class directly.